### PR TITLE
[TASK] Streamline database expression builder chapter

### PR DESCRIPTION
**Pull request by brotkrueml**
Releases: main, 11.5

### CODE REVIEW

File: Documentation/ApiOverview/Database/ExpressionBuilder/Index.rst [hunk 1]
```diff
@@ -1,298 +1,336 @@
...
+..  code-block:: php
+    :caption: EXT:my_extension/Classes/Domain/Repository/MyTableRepository.php
 
-* :php:`->or()` disjunction
+    $rows = $this->connectionPool->getQueryBuilderForTable('tt_content')
+        ->select('uid', 'header', 'bodytext')
+        ->from('tt_content')
+        ->where(
+            // `bodytext` = 'klaus' AND `header` = 'peter'
+            $queryBuilder->expr()->eq('bodytext', $queryBuilder->createNamedParameter('klaus')),
```
Comment (linawolf) [inline Documentation/ApiOverview/Database/ExpressionBuilder/Index.rst:53]: Where does `$queryBuilder` come from?

File: Documentation/ApiOverview/Database/ExpressionBuilder/Index.rst [hunk 2]
```diff
@@ -1,298 +1,336 @@
...
 
-* All methods that have a :php:`$value` or :php:`$valueList` as second argument **must** be quoted, usually by calling
-  :ref:`$queryBuilder->createNamedParameter() <database-query-builder-create-named-parameter>` or
-  :ref:`$queryBuilder->quoteIdentifier() <database-query-builder-quote-identifier>`. **Failing to do so will end
-  up in SQL injections!**
+*   All methods that have a :php:`$value` or :php:`$valueList` as second
+    argument **must** be quoted, usually by calling
+    :ref:`$queryBuilder->createNamedParameter() <database-query-builder-create-named-parameter>`
+    or :ref:`$queryBuilder->quoteIdentifier() <database-query-builder-quote-identifier>`.
+    **Failing to do so will end up in SQL injections!**
```
Comment (linawolf) [inline Documentation/ApiOverview/Database/ExpressionBuilder/Index.rst:164]:
```suggestion
    **Failing to do so introduces security risks :ref:`SQL injections <security-sql-injection`!**
```

File: Documentation/ApiOverview/Database/ExpressionBuilder/Index.rst [hunk 3]
```diff
@@ -1,298 +1,336 @@
...
-=========
+    $expressionBuilder = $queryBuilder->expr();
 
-.. versionchanged:: 11.5.10
-   The methods :php:`andX()` and :php:`orX()` are deprecated and replaced
-   by :php:`and()` and :php:`or()` to match with `doctrine/dbal`, which
-   `deprecated <https://github.com/doctrine/dbal/commit/84328cd947706210caebcaea3ca0394b3ebc4673>`_
-   these methods.
+It is good practice not to assign an instance of the :php:`ExpressionBuilder` to
+a variable, but to use it directly within the code flow of the query builder
```
[the remark this answers is not in the document]
Comment (brotkrueml) [reply Documentation/ApiOverview/Database/ExpressionBuilder/Index.rst:42]: The expression builder can only be used in a query builder where method. So to use it you have already somewhere a query builder retrieved.

File: Documentation/ApiOverview/Database/ExpressionBuilder/Index.rst (hunk 1 above)
Comment (brotkrueml) [reply Documentation/ApiOverview/Database/ExpressionBuilder/Index.rst:53]: Yup, you are right, that is inconsistent.

File: Documentation/ApiOverview/Database/ExpressionBuilder/Index.rst (hunk 3 above)
[the remark this answers is not in the document]
Comment (linawolf) [reply Documentation/ApiOverview/Database/ExpressionBuilder/Index.rst:42]: Ah, duh! Expression builder not query builder, sorry

**Review by linawolf — APPROVED** [review]

**Pull request merged**